projects/plugins/jetpack/modules/shortcodes/spotify.php

```diff
@@ -48,7 +48,7 @@ function jetpack_spotify_shortcode( $atts = array(), $content = '' ) {
 	// Spotify accepts both URLs and their Spotify ID format, so let them sort it out and validate.
 	$embed_url = add_query_arg( 'uri', rawurlencode( $id ), 'https://embed.spotify.com/' );
 
-	return '<iframe src="' . esc_url( $embed_url ) . '" style="display:block; margin:0 auto; width:' . esc_attr( $atts['width'] ) . 'px; height:' . esc_attr( $atts['height'] ) . 'px;" frameborder="0" allowtransparency="true"></iframe>';
+	return '<iframe src="' . esc_url( $embed_url ) . '" style="display:block; margin:0 auto; width:' . esc_attr( $atts['width'] ) . 'px; height:' . esc_attr( $atts['height'] ) . 'px;" frameborder="0" allowtransparency="true" loading="lazy"></iframe>';
```

Comment:
Looking into this, I do not think we need to add the attribute ourselves. WordPress itself will add the lazy loading attribute to all iFrames in the post content. This was introduced in WordPress 5.7, in this ticket:
https://core.trac.wordpress.org/ticket/50756

For this to work, however, the iFrame needs to have `width` and `height` attributes set. That's currently not the case with our Spotify iFrame (we only set dimensions in the inline styles). I would suggest trying that instead, so we can let WordPress handle lazy loading instead of doing it ourselves.

Something like this:

```suggestion
	return sprintf(
		'<iframe src="%1$s" width="%2$d" height="%3$d" style="display:block; margin:0 auto; width:%2$dpx; height:%3$dpx;" frameborder="0" allowtransparency="true"></iframe>',
		esc_url( $embed_url ),
		esc_attr( $atts['width'] ),
		esc_attr( $atts['height'] )
	);
```

If we do this, the test you've added will not be necessary anymore. You could, however, add a new test for the width and height attributes.

Reply:
Hi @jeherve, I tried your code suggestion. Looks like `wp_iframe_tag_add_loading_attr ` is not able to add `loading` attribute. I tried two approaches:

1. Code suggestion by you
![Screenshot 2021-08-10 19:16:06](https://user-images.githubusercontent.com/21127788/128880223-468fba81-a6c3-47ab-8d1c-973315f3975d.png)

2. apply filter `wp_iframe_tag_add_loading_attr` and then return final url.
![Screenshot 2021-08-10 19:26:28](https://user-images.githubusercontent.com/21127788/128880290-98a50366-b894-4baf-b6f5-64919427643b.png)

Let me know if I'm doing something wrong/missing here. I went through the link of the [PR](https://core.trac.wordpress.org/ticket/50756)
I read all the comments and am not sure whether it is relevant or not.
![Screenshot 2021-08-10 19:31:14](https://user-images.githubusercontent.com/21127788/128880764-0bdfea13-8036-4366-8253-e849366afcc4.png)

Reply:
Ah, I see now. It would seem that WordPress Core won't add the attribute to shortcodes built manually like this. We would have to rely on `$wp_embed;` and `$wp_embed->shortcode` instead, like we do for the Kickstarter shortcode for example:

https://github.com/Automattic/jetpack/blob/26db3e436ccca3e16a62d02d95e792a81f38a1e4/projects/plugins/jetpack/modules/shortcodes/kickstarter.php#L33-L34

cc @kraftbj who did some work on #20060 recently, and who may have more insight on this. 

For now the simplest solution may be to add the attribute manually like you did.